streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamPrinter.java

```diff
@@ -0,0 +1,126 @@
+import org.apache.kafka.streams.processor.ProcessorSupplier;
+
+import java.io.PrintStream;
+
+
+class KStreamPrinter<K, V> implements ProcessorSupplier<K, V> {
+
+    private final PrintStream printStream;
+    private Serde<?> keySerde;
+    private Serde<?> valueSerde;
+    private static final String NULL_KEY = "null_key";
+    private boolean notStandardOut;
+
+
+    KStreamPrinter(PrintStream printStream, Serde<?> keySerde, Serde<?> valueSerde) {
+        this.keySerde = keySerde;
+        this.valueSerde = valueSerde;
+        if (printStream == null) {
+            this.printStream = System.out;
+        } else {
+            this.printStream = printStream;
+            notStandardOut = true;
+        }
+    }
+
+    KStreamPrinter(PrintStream printStream) {
+        this(printStream, null, null);
+    }
+
+    KStreamPrinter(Serde<?> keySerde, Serde<?> valueSerde) {
+        this(null, keySerde, valueSerde);
+    }
+
+    KStreamPrinter() {
+        this(null, null, null);
+    }
+
+    @Override
+    public Processor<K, V> get() {
+        return new KStreamPrinterProcessor(this.printStream, this.keySerde, this.valueSerde);
+    }
+
+
+    private class KStreamPrinterProcessor extends AbstractProcessor<K, V> {
+        private final PrintStream printStream;
+        private Serde<?> keySerde;
+        private Serde<?> valueSerde;
+        private ProcessorContext processorContext;
+
+        private KStreamPrinterProcessor(PrintStream printStream, Serde<?> keySerde, Serde<?> valueSerde) {
+            this.printStream = printStream;
+            this.keySerde = keySerde;
+            this.valueSerde = valueSerde;
+        }
+
+        @Override
+        public void init(ProcessorContext context) {
+            this.processorContext = context;
+
+            if (this.keySerde == null) {
+                keySerde = this.processorContext.keySerde();
+            }
+
+            if (this.valueSerde == null) {
+                valueSerde = this.processorContext.valueSerde();
+            }
+        }
+
+        @Override
+        public void process(K key, V value) {
+            Object keyToPrint = maybeDeserialize(key, keySerde.deserializer());
+            Object valueToPrint = maybeDeserialize(value, valueSerde.deserializer());
+
+            String stringKey = keyToPrint == null ? NULL_KEY : keyToPrint.toString();
+
+            printStream.println(stringKey + " , " + valueToPrint);
+
+            this.processorContext.forward(key, value);
+        }
+
+
+        private Object maybeDeserialize(Object receivedElement, Deserializer<?> deserializer) {
+            if (receivedElement == null) {
+                return null;
+            }
+
+            if (receivedElement instanceof byte[]) {
+                //TODO  calling this.processorContext.topic() throws Exception in unit test
+                return deserializer.deserialize("", (byte[]) receivedElement);
+            }
+
```

Comment:
TODO will be removed, left to force the issue with myself that MockProcessorContext.topic() throws UnsupportedOperationException but I wanted to push the PR with unit tests. Should the MockProcessorContext.topic() method be updated to return default or empty string?